### DISCUSSION TITLE
How to make a custom Interface for Date input

### DISCUSSION THREAD

**Discussion by Pachat** [directus/directus#15541] [Archive]
I'd like to make a custom Interface to allow the user to type the date directly without a calendar.
_The feature has been discussed in  #3728 but is not yet, if it is some day, implemented in Directus._

So, I tried to make an appropriate extension.
The problem is: what to put in `src/index.js` as `types`? I tried `date`, `datetime`, `timestamp` but none of them would show as Interface of a `Datetime` field in Directus.

(I don't want to use "string" as the field has to be a timestamp in the DB, PostgreSQL here)

**Comment by rijkvanzanten** [comment]
The `types` you mentioned are correct. You can also take a look at the system interfaces to reference what they're using:

https://github.com/directus/directus/blob/b1106e183df82e247cb4f49209cc0e0d78372851/app/src/interfaces/datetime/index.ts#L11-L12

> but none of them would show as Interface of a Datetime field in Directus.

What does your full index file look like so far? Where are you (not) seeing this?

**Reply by Pachat** [reply]
Here is the full new `src/index.js`
```
import InterfaceComponent from './interface.vue';

export default {
	id: 'simpledate',
	name: 'Date',
	icon: 'box',
	description: 'Date DD/MM/YYYY',
	component: InterfaceComponent,
	options: null,
	types: ['dateTime', 'date', 'time', 'timestamp'],
	group: 'selection',
};

```

And now it works!
Thanks :)
